src/model/serde_config.rs

```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Deserializer};
+use serde_json::Value;
+
+pub fn bool_from_int_or_bool<'de, D>(deserializer: D) -> Result<Option<bool>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    match Value::deserialize(deserializer)? {
+        Value::Bool(b) => Ok(Some(b)),
+        Value::Number(n) => {
+            if let Some(i) = n.as_i64() {
+                match i {
+                    0 => Ok(Some(false)),
+                    1 => Ok(Some(true)),
+                    _ => Err(serde::de::Error::custom(
+                        "expected 0 or 1 for optional boolean field",
+                    )),
+                }
+            } else {
+                Err(serde::de::Error::custom(
+                    "expected an integer for optional boolean field",
```

Comment:
```suggestion
        Value::Number(n) => {

             let Some(i) = n.as_i64() else {
                 return Err(serde::de::Error::custom(
                    "expected an integer for optional boolean field"));
}

match i {

 0 => Ok(Some(false)),
 1 => Ok(Some(true)),
 _ => Err(serde::de::Error::custom(
                        "expected 0 or 1 for optional boolean field"
}
```

Maybe there is some format or brackets issue but you get the idea , it is better to use `let else` expression, because it allows you to do an early return and you have less nested code.

Reply:
https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html

Reply:
fixed [Implemented suggested change](https://github.com/ThalosES/vinted-rs/pull/118/commits/edd36c2ffb6878c5d9da8da536ab258c3cc615f6)